### PR TITLE
Reorganization ruby and amqp workers

### PR DESCRIPTION
**Pull request by mnaichuk**

### CODE REVIEW

File: app/worker/daemons/k.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+#   - Check business logic.
+#   - Move to KLineService.
+#   - Add appropriate specs.
+
+# K-line point is represented as array of 5 numbers:
+# [timestamp, open_price, max_price, min_price, last_price, period_volume]
+
+module Worker
+  module Daemons
+    class K < Worker::Daemons::Base
```
Comment (ysv) [inline app/worker/daemons/k.rb]:
```ruby
class K < Base
```

File: app/worker/daemons/base.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      end
+
+      def stop
+        @running = false
+      end
+
+      def run
+        while running
+          begin
+            process
```
Comment (ysv) [inline app/worker/daemons/base.rb]: I would prefer to put sleep here

**Review by ysv — CHANGES_REQUESTED** [review]
It's better to use Workers as a module name

File: lib/daemons/daemons.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# frozen_string_literal: true
+
+require File.join(ENV.fetch('RAILS_ROOT'), 'config', 'environment')
+
+raise "Worker name must be provided." if ARGV.size == 0
+
+running = true
+terminate = proc do
+  puts "Terminating worker .."
+  @worker.stop
```
Comment (ysv) [inline lib/daemons/daemons.rb]: Initialize worker before it's usage
Don't use instance variable for it '@'

**Review by ysv — CHANGES_REQUESTED** [review]

File: lib/daemons/daemons.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+
+require File.join(ENV.fetch('RAILS_ROOT'), 'config', 'environment')
+
+raise "Worker name must be provided." if ARGV.size == 0
+
+name = ARGV[0]
+running = true
+worker = "Workers::Daemons::#{name.camelize}".constantize.new(running)
```
Comment (ysv) [inline lib/daemons/daemons.rb]: remove running from initialization

**Comment by ysv** [comment]
Don't forget to update god.
Check if any doc need updates I would prefer the next:
1. Rename daemons.md -> workers.md (so we have directory and doc naming convention)
2. Prefix all daemons in daemons.md with `daemon` so we will have something like:
```md
## amqp:deposit_collection

This daemon transfer incoming deposits from deposit wallet to withdraw wallets (hot, warm, cold).

## daemon:withdraw_audit

This daemon validates withdrawals and sends them to amqp:withdraw_coin.
```

**Review by ysv — APPROVED** [review]

**Review by mod — APPROVED** [review]

**Pull request merged**